isthmus/src/test/java/io/substrait/isthmus/Substrait2SqlTest.java

```diff
@@ -42,7 +42,78 @@ public void simpleTestDecimal() throws Exception {
         "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0 from lineitem where l_shipdate < date '1998-01-01' ");
   }
 
+  @Test
+  public void simpleJoin() throws Exception {
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem left join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem right join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem full join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+  }
+
+  @Test
+  public void simpleTestAgg() throws Exception {
+    test(
+        "select l_partkey, count(l_tax), COUNT(distinct l_discount) from lineitem group by l_partkey");
+  }
+
+  @Test
+  public void simpleTestAggNoGB() throws Exception {
+    test("select count(l_tax), count(distinct l_discount) from lineitem");
```

Comment:
Can we add a test that uses `approx_count_distinct`? Something like this:
```
test("select approx_count_distinct(l_discount) from lineitem");
```

Reply:
for  `select approx_count_distinct(l_discount) from lineitem"`,  from sql -> calcite rel, we have approximate/distinct for count() function, but after calcite rel -> substrait rel , the `approximate` flag is lost during `SubstraitRelVisitor`.

Looks like we have to fix the issue calcite rel -> substrait rel.

Reply:
Does it work going the other direction (from substrait plan -> sql)? You can generate a substrait plan using `count(distinct l_discount)` and then run this code to replace them with `approx_count_distinct`:
```
    ReplaceCountDistinctWithApprox action = new ReplaceCountDistinctWithApprox();
    Plan newPlan = action.modify(oldPlan).orElse(oldPlan);
```
Finally, see if your code to go from substrait plan to sql writes the correct sql with the `approx_count_distinct` call.